src/utils.jl

```diff
@@ -470,3 +475,42 @@ function ic_phase_conversion(
         end
     end
 end
+
+"""
+    check_for_duplicates(log_volume_km3_vector::Union{Float64,Vector{Float64}}, InitialConc_H2O_vector::Union{Float64,Vector{Float64}}, InitialConc_CO2_vector::Union{Float64,Vector{Float64}}, log_vfr_vector::Union{Float64,Vector{Float64}}, depth_vector::Union{Float64,Vector{Float64}})::Nothing
+
+This function checks if any of the input arrays contain duplicate elements. If duplicates are found, it raises an error indicating which arguments have duplicates.
+- This function is used within the `chamber` function
+"""
+function check_for_duplicates(
+    log_volume_km3_vector::Union{Float64,Vector{Float64}},
+    InitialConc_H2O_vector::Union{Float64,Vector{Float64}},
+    InitialConc_CO2_vector::Union{Float64,Vector{Float64}},
+    log_vfr_vector::Union{Float64,Vector{Float64}},
+    depth_vector::Union{Float64,Vector{Float64}},
+)::Nothing
+    duplicate_arguments = []
+    argument_names = [
+        "log_volume_km3", "InitialConc_H2O", "InitialConc_CO2", "log_vfr", "depth"
+    ]
+
+    for (i, array) in enumerate([
+        log_volume_km3_vector,
+        InitialConc_H2O_vector,
+        InitialConc_CO2_vector,
+        log_vfr_vector,
+        depth_vector,
+    ])
+        if length(unique(array)) != length(array)
+            push!(duplicate_arguments, argument_names[i])
+        end
+    end
+
+    if !isempty(duplicate_arguments)
+        joined_arguments = join(duplicate_arguments, ", ")
+        error(
+            "The following arguments contain duplicates: $joined_arguments. Please remove the duplicate elements and submit again.",
+        )
+    end
```

Comment:
```suggestion
    argument_names = [
        "log_volume_km3", "InitialConc_H2O", "InitialConc_CO2", "log_vfr", "depth"
    ]
    reps = [argument_names[i] for (i, array) in enumerate([
        log_volume_km3_vector,
        InitialConc_H2O_vector,
        InitialConc_CO2_vector,
        log_vfr_vector,
        depth_vector,
    ]) if length(unique(array)) != length(array)]

    !isempty(reps) && error(
        "The following arguments contain duplicates: $(join(reps, ", ")). Please remove the duplicate elements and submit again.",
    )
```

Consider using an array comprehension -- avoids dynamically growing an array.